### ISSUE TITLE
Task.Run doesn't run on a separate thread.

### ISSUE THREAD

**Issue by FinVamp1** [Azure/azure-webjobs-sdk-script#2248]
#### Investigative information

Please provide the following:

- Timestamp: 22nd of December 2017 14:00 to 15:00
- Function App version (1.0 or 2.0-beta):
- Function App name: finxldpvvv4d772y-triggers 
- Function name(s) (as appropriate):HttpTriggerCSharp1 
- Invocation ID: 44018b6a-ef8b-4368-8589-c3b4ba8a1e32
- Region:South Central US.

#### Repro steps

Provide the steps required to reproduce the problem:

Example: 

1. Create a new Function App.
2. Create a new HTTP Trigger C# Function.
3. Use code that calls Task.Run. (Sample below)
4. Run the Function

#### Expected behavior

Provide a description of the expected behavior.

After Step 4 the Task should run on a different thread.

E.G

2017-12-22T14:50:14  Welcome, you are now connected to log-streaming service.
2017-12-22T14:50:26.610 Function started (Id=44018b6a-ef8b-4368-8589-c3b4ba8a1e32)
2017-12-22T14:50:26.954 Start original Thread 37
2017-12-22T14:50:26.954 Start new Thread 21
2017-12-22T14:50:31.968 Finish new Thread 21
2017-12-22T14:50:31.968 Finish original Thread 37
2017-12-22T14:50:32.000 Function completed (Success, Id=44018b6a-ef8b-4368-8589-c3b4ba8a1e32, Duration=5381ms)

#### Actual behavior

Provide a description of the actual behavior observed.

The Task.Run executes on the Same Thread.

2017-12-22T14:50:14  Welcome, you are now connected to log-streaming service.
2017-12-22T14:50:26.610 Function started (Id=44018b6a-ef8b-4368-8589-c3b4ba8a1e32)
2017-12-22T14:50:26.954 Start original Thread 37
2017-12-22T14:50:26.954 Start new Thread 37
2017-12-22T14:50:31.968 Finish new Thread 37
2017-12-22T14:50:31.968 Finish original Thread 37
2017-12-22T14:50:32.000 Function completed (Success, Id=44018b6a-ef8b-4368-8589-c3b4ba8a1e32, Duration=5381ms)

#### Known workarounds

Provide a description of any known workarounds.

No workarounds exist.

#### Related information 

Provide any related information 

* Programming language used 
C#

<details>
<summary>Source</summary>

```csharp

using System.Net;
using System.Threading;
using System.Threading.Tasks;

public static HttpResponseMessage Run(HttpRequestMessage req, TraceWriter log)
{
    log.Info("Start original Thread " + Thread.CurrentThread.ManagedThreadId.ToString());
    var t = Task.Run(() =>{
        log.Info("Start new Thread " + Thread.CurrentThread.ManagedThreadId.ToString());
        System.Threading.Thread.Sleep(5000);
        log.Info("Finish new Thread " + Thread.CurrentThread.ManagedThreadId.ToString());
    });
    t.Wait();
    log.Info("Finish original Thread " + Thread.CurrentThread.ManagedThreadId.ToString());

    return req.CreateResponse(HttpStatusCode.OK, "Hello World");
}


</details>

**Comment by dallancarr** [comment]
This dependent on the TaskScheduler and it's preferred way to execute a given task. Generally speaking you have no control over thread creation but can provide hints. But you wouldn't want to do this without prior knowledge of th thread pool.

**Comment by FinVamp1** [comment]
Thanks @dallancarr I think this behaviour differs from the Console experience where you get another thread straight away from the scheduler. I agree with your points though and executing on the same thread doesn't always indicate a performance issue.

**Comment by dallancarr** [comment]
I imagine the function is already running on a background thread. 😊

**Comment by FinVamp1** [comment]
Discussed this further with engineering. 

Task.Run does not have threading assurances with the default ASP.NET/Functions scheduler
A new task execution doesn't mean a new (or other) thread... 

The customer should remain assured that the Tasks will run in the order they defined using Task.Run or they can move to making the Function Async and return a Task.
https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-csharp#Async 

So this is by design.
